src/main/java/io/jenkins/plugins/thememanager/ThemeManagerFactoryDescriptor.java

```diff
@@ -37,6 +38,29 @@ public ThemeManagerFactory getInstance() {
    */
   public abstract String getThemeId();
 
+  /**
+   * A unique key for a theme plugin
+   *
+   * <p>This should be all lower case and URL safe, i.e. 'dark', 'neo2'
+   *
+   * <p>Used in the html dataset namespace for the theme.
+   * @return the theme key
+   */
+  public abstract String getThemeKey();
+
+  /**
+   * If the theme's CSS will only be selected under the '[data-theme=theme-id]' selector
+   *
+   * It will be served on all pages even if not activated, ensure all selectors are behind the dataset id.
```

Comment:
Is `dataset id` the `theme-id`?

Reply:
got slightly refactored from ID because there was already an ID which didn't have to be unique inside a plugin, e.g. solarized was allowed to handle all 3 themes, so I've added a 'key'
```suggestion
   * It will be served on all pages even if not activated, ensure all selectors are behind the dataset selector.
```